contentctl/actions/test.py

```diff
@@ -46,29 +46,31 @@ class TestInputDto:
 class Test:
 
     def filter_detections(self, input_dto: TestInputDto)->TestInputDto:
-        
         if not input_dto.config.enable_integration_testing:
-            #Skip all integraiton tests if integration testing is not enabled:
+            # Skip all integraiton tests if integration testing is not enabled:
             for detection in input_dto.detections:
                 for test in detection.tests:
                     if isinstance(test, IntegrationTest):
                         test.skip("TEST SKIPPED: Skipping all integration tests")
-        
-        list_after_filtering:List[Detection] = []
-        #extra filtering which may be removed/modified in the future
-        for detection in input_dto.detections:
```

Comment:
Would we rather skip these detections here instead of during model creation (see `adjust_tests_and_groups` in `detection_abstract`)?

Reply:
No, I think we should do that during model creation in `adjust_tests_and_groups`. In my PR (that I will link to) I will completely remove this function. I will leave this open until I can leave the link for your review.

Reply:
Completely removed here: https://github.com/splunk/contentctl/pull/260/files#diff-8d0f6a0b0a6ce73f573c88af4091f8162dc4c20424beec4c340cece456ceb029L48-L57
and the call to it:
https://github.com/splunk/contentctl/pull/260/files#diff-12b4d2896c61f39f042922078f1a4239335c75e1c7aed3614c40e271b00477fdL117-L120

Reply:
Agreed. There's on caveat that I'm calling out on your PR

Reply:
Resolving this conversation given that we discussed it and made changes in the linked PR.
In summary, we actually kept this filtering in the same spot so that we can support test creation during the validate and build modes, not JUST the test mode.